R/corrpart.b.R

```diff
@@ -246,72 +233,59 @@ corrPartClass <- R6::R6Class(
+        .test = function(var1, var2, controls, type, hyp) {
 
             data <- private$.cleanData(var1, var2)
             var1 <- data[[var1]]
             var2 <- data[[var2]]
 
-            if (type != 'zero' && length(controls) > 0)
-                controls <- data[, controls]
-
-            results <- list()
+            n <- length(data[[1]])
+            d <- 1 + as.integer(type == 'semi')
+            c <- length(controls)
+            if (c == 0) { rdta <- cbind(var1, var2) } else { rdta = cbind(var1, var2, data[, controls]) }
+            results <- replicate(d, list(r = NaN, rp = 1, rho = NaN, rhop = 1, tau = NaN, taup = 1, n = n))
 
             suppressWarnings({
-
-                if (type == 'zero' || length(controls) == 0) {
-
-                    res1 <- try(cor.test(var1, var2, method='pearson'))
-                    res2 <- try(cor.test(var1, var2, method='spearman'))
-                    res3 <- list()
-                    if (self$options$kendall)
-                        res3 <- try(cor.test(var1, var2, method='kendall'))
-
-                } else if (type == "part") {
-
-                    res1 <- try(ppcor::pcor.test(var1, var2, controls, method='pearson'))
-                    res2 <- try(ppcor::pcor.test(var1, var2, controls, method='spearman'))
-                    res3 <- list()
-                    if (self$options$kendall)
-                        res3 <- try(ppcor::pcor.test(var1, var2, controls, method='kendall'))
-
-                } else {
-
-                    res1 <- try(ppcor::spcor.test(var1, var2, controls, method='pearson'))
-                    res2 <- try(ppcor::spcor.test(var1, var2, controls, method='spearman'))
-                    res3 <- list()
-                    if (self$options$kendall)
-                        res3 <- try(ppcor::spcor.test(var1, var2, controls, method='kendall'))
-
-                }
-
-                if ( ! base::inherits(res1, 'try-error')) {
-                    results$r  <- res1$estimate
-                    results$rp <- res1$p.value
-                } else {
-                    results$r <- NaN
-                    results$rp <- NaN
-                }
-
-                if ( ! base::inherits(res2, 'try-error')) {
-                    results$rho <- res2$estimate
-                    results$rhop <- res2$p.value
-                } else {
-                    results$rho <- NaN
-                    results$rhop <- NaN
-                }
-
-                if ( ! base::inherits(res3, 'try-error')) {
-                    results$tau <- res3$estimate
-                    results$taup <- res3$p.value
-                } else {
-                    results$tau <- NaN
-                    results$taup <- NaN
-                }
-
-                results$n <- length(data[[1]])
+                for (method in c('pearson', 'spearman', 'kendall')) {
+                    # calculate 'kendall' only if the respective tick box is set
```

Comment:
Try to use comments sparsely. Comments should only be used to describe general functionality of a block of code, not the exactly functionality of a line of code. For instance, the if statement is pretty verbose over here so you're basically saying the same thing twice. For more info, see for example https://dev.to/codemouse92/to-comment-or-not-to-comment-3f7h